api/migrations/20200603153526-bcmiDataLoad.js

```diff
@@ -0,0 +1,146 @@
+    const publishedMines = await getRequest(bcmiUrl + '/api/projects/published');
+    console.log('Located ' + publishedMines.length + ' mines. Batching updates...');
+
+    const promises = [];
+    // build up a collection of all requests
+    for (let i = 0; i < publishedMines.length; i++) {
+      try {
+        // The published endpoint doesn't have links, refresh the mineData object
+        let mineData = await getRequest(bcmiUrl + '/api/project/bycode/' + publishedMines[i].code);
+        promises.push(updateMine(mineData, nrpti));
+      } catch(err) {
+        console.error('Could not find ' + publishedMines[i]._id + ' : ' + publishedMines[i].name);
+        // dont rethrow, we'll just ignore this one as a failure and check the rest
+      }
+    }
+
+    // fire off the requests and wait
+    let results = await Promise.all(promises);
+
+    let updatedCount = 0;
+    let notFoundCount = 0;
+    results.forEach(result => {
+      if (Object.prototype.hasOwnProperty.call(result, 'notfound')) {
+        notFoundCount++;
+        console.log('Could not find ' + result.data._id + ' : ' + result.data.name);
+      } else {
+        updatedCount++;
+      }
+    });
+
+    // we're done
+    console.log('BCMI migration complete.');
+    console.log('Of ' + publishedMines.length + ' mines in BCMI, ' + notFoundCount + ' could not be found in NRPTI, and ' + updatedCount + ' were updated.');
+  } catch(err) {
+    console.error('Error on BCMI dataload: ' + err);
+  }
+
+  mClient.close();
+};
+
+exports.down = function(db) {
+  return null;
+};
+
+exports._meta = {
+  "version": 1
+};
+
+async function updateMine(mineData, nrpti) {
+  let nrptiMines = await nrpti.find({ _schemaName: 'MineBCMI', name: mineData.name}).toArray();
+  // should have 1 result returned. Any more or less, just ignore this update
+  if (nrptiMines.length === 1) {
+    let externalLinks = [];
+    // format the links from the data
+    for(const idx in mineData.externalLinks) {
+      const link = mineData.externalLinks[idx];
+      externalLinks.push(link.link);
+    }
+
+    return nrpti.update({ _id: new ObjectID(nrptiMines[0]._id) }, {
+      $set: {
+        type:            mineData.type,
+        summary:         mineData.description, // BCMI doesn't have a "summary" attribute
+        description:     mineData.description,
+        links:           externalLinks,
+        updatedBy:       'NRPTI BCMI Data Migration',
+        sourceSystemRef: 'mem-admin'
+      }
+    });
+
+    // Alternatively...
+    /* let nrptiMine = await require('../src/models/bcmi/mine').findById(nrptiMines[0]._id);
+
+    nrptiMine.type        = mineData.type;
+    nrptiMine.summary     = mineData.description;
+    nrptiMine.description = mineData.description;
+    nrptiMine.links       = externalLinks;
+    nrptiMine.updatedBy   = 'NRPTI BCMI Data Migration';
+
+    return nrptiMine.save(); */
```

Comment:
May as well remove, if we aren't going this route.

Reply:
Yep, forgot to clean that out. It's been removed